### PR TITLE
feat(59): invitations

### PR DESCRIPTION
**Pull request by dartwlad**

### CODE REVIEW

File: packages/server/src/api/index.ts [hunk 1]
```diff
@@ -17,7 +17,10 @@ const userManagementUserUpdate = `${management}/user-update`;
 const userManagementMergeWithSocial = `${management}/merge-with-social`;
 const userManagementAddFacebook = `${management}/add-facebook`;
 const userManagementAddGoogle = `${management}/add-google`;
+const userManagementShelfInvites = `${management}/user-invites`;
 const shelfUserInvite = `${shelf}/user-invite`;
+const shelfPendingInvites = `${shelf}/pending-invites`;
+const shelfRevokeInvite = `${shelf}/revoke-invite`;
 const shelfUserInviteAccept = `${shelf}/user-invite-accept`;
```
Comment (dubbha) [inline packages/server/src/api/index.ts:24]: I called these `user-invite` and `user-invite-accept`, so would expect these to follow the same pattern like `user-invite-revoke`. If you don't like my naming you could change `user-invite-accept` instead, or at least put the new two after it, not between `user-invite` and `user-invite-accept`.

File: packages/server/src/shelf/shelf.controller.ts [hunk 2]
```diff
@@ -106,10 +106,24 @@ export default class ShelfController extends BaseController {
...
+    this.router.delete(
+      this.serverApi.shelfRevokeInvite,
+      authMiddleware,
+      shelfAdminMiddleware,
+      this.shelfRevokeInviteHandler
+    );
   }
 
-  private userInviteHanlder = async (
+  private userInviteHandler = async (
```
Comment (dubbha) [inline packages/server/src/shelf/shelf.controller.ts:126]: My bad, guess I was sleepy. Same for the `actionHanlder` by the way.

File: packages/server/src/shelf/shelf.controller.ts [hunk 3]
```diff
@@ -106,10 +106,24 @@ export default class ShelfController extends BaseController {
       this.serverApi.shelfGetLastActions,
       authMiddleware,
       this.getLastActionsHanlder
-    )
+    );
+
+    this.router.post(
```
Comment (dubbha) [inline packages/server/src/shelf/shelf.controller.ts:111]: Following our conversation today maybe we could start using `GET`s to get things.

File: packages/server/src/user-management/user-management.controller.ts [hunk 4]
```diff
@@ -51,6 +51,12 @@ export default class UserManagementController extends BaseController {
       this.providersAuthService.verifyGoogleLogin,
       this.mergeUserProfilesHandler
     );
+
+    this.router.get(
```
Comment (dubbha) [inline packages/server/src/user-management/user-management.controller.ts:55]: ...like we do here :)
BTW, did you check that `axios.get` would send auth token header just the same?

**Review by dubbha — APPROVED** [review]
See my comments, fine otherwise

File: packages/server/src/shelf/shelf.controller.ts (hunk 3 above)
Comment (dartwlad) [reply packages/server/src/shelf/shelf.controller.ts:111]: Check sluck

File: packages/server/src/user-management/user-management.controller.ts (hunk 4 above)
Comment (dartwlad) [reply packages/server/src/user-management/user-management.controller.ts:55]: Yes

**Review by 0to1to0to1 — APPROVED** [review]

**Pull request merged**